### PR TITLE
oauth-apiserver: add access token inactivity timeout observer

### PR DESCRIPTION
**Pull request by stlaz**
/assign @sttts 
cc @ingvagabund

### CODE REVIEW

**Comment by stlaz** [comment]
The operator tests failed because the default test timeout was 10m but the token timeout tests are tremendously time consuming.

I increased the token timeout period and attempted to decrease the time per the test by roughly 15minutes.

**Comment by stlaz** [comment]
/retest
infra

**Comment by stlaz** [comment]
/retest
install failed on e2e-operator

**Comment by stlaz** [comment]
/retest
no gcp resources

File: test/e2e/tokentimeout_test.go [hunk 1]
```diff
@@ -95,7 +95,7 @@ func TestTokenInactivityTimeout(t *testing.T) {
 		testTokenValidity(t, req, http.StatusOK, tokenWithoutTimeout.Name, "token with out timeout should work")
 	}
 
-	configInactivityTimeout := int32(600)
+	configInactivityTimeout := int32(420)
```
Comment (sttts) [inline test/e2e/tokentimeout_test.go:99]: why still that long? Why not `O(5 minutes)` ?

File: test/e2e/tokentimeout_test.go [hunk 2]
```diff
@@ -299,3 +305,19 @@ func createOAuthClient(t *testing.T, oauthClientClient *oauthclient.OauthV1Clien
...
+	t.Logf("waiting all oauth-apiserver replicas to be updated and ready")
+
+	err := wait.PollImmediate(time.Second, 10*time.Minute, func() (bool, error) {
+		deployment, err := kubeClient.AppsV1().Deployments("openshift-oauth-apiserver").Get(context.Background(), "apiserver", metav1.GetOptions{})
+		if err != nil {
+			t.Logf("failed to retrieve oauth-apiserver's deployment: %v", err)
+			return false, nil
+		}
+
+		return deployment.Status.Replicas == deployment.Status.ReadyReplicas && deployment.Status.UpdatedReplicas == deployment.Status.Replicas, nil
```
Comment (sttts) [inline test/e2e/tokentimeout_test.go:319]: not spec.replicas?

File: test/e2e/tokentimeout_test.go (hunk 1 above)
Comment (sttts) [reply test/e2e/tokentimeout_test.go:99]: Or iow: what's the reason for these two numbers?

Comment (stlaz) [reply test/e2e/tokentimeout_test.go:99]: 300s is the minimal, I think the original test was trying to prove that one of the timeouts apply and it's not the other one, so it had different values. I thought 2 minutes should be a sufficient-enough difference.

**Pull request merged**